vm/systemSmartContracts/auction.go

```diff
@@ -631,6 +631,8 @@ func (s *stakingAuctionSC) stake(args *vmcommon.ContractCallInput) vmcommon.Retu
 			if len(args.Arguments[i]) == len(args.CallerAddr) {
 				if !isAlreadyRegistered {
 					registrationData.RewardAddress = args.Arguments[i]
```

Comment:
there is only one registrationData.RewardAddress, but you are iterating through "i" rewardAddresses. Only the last one will be saved, is this the intended behaviour?